beluga/test/beluga/views/test_take_evenly.cpp

```diff
@@ -15,84 +15,160 @@
 #include <gmock/gmock.h>
 #include <gtest/gtest.h>
 
+#include <list>
 #include <vector>
 
 #include <range/v3/range/conversion.hpp>
 #include <range/v3/view/iota.hpp>
+#include <range/v3/view/reverse.hpp>
 
 #include "beluga/views/take_evenly.hpp"
 
 namespace {
 
+TEST(TakeEvenlyView, ConceptChecksFromContiguousRange) {
+  auto input = std::array{1, 2, 3, 4, 5, 6, 7, 8, 9, 10};
+  auto output = beluga::views::take_evenly(input, 2);
+
+  static_assert(ranges::common_range<decltype(input)>);
+  static_assert(!ranges::common_range<decltype(output)>);
+
+  static_assert(!ranges::viewable_range<decltype(input)>);
+  static_assert(ranges::viewable_range<decltype(output)>);
+
+  static_assert(ranges::forward_range<decltype(input)>);
+  static_assert(ranges::forward_range<decltype(output)>);
+
+  static_assert(ranges::sized_range<decltype(input)>);
+  static_assert(ranges::sized_range<decltype(output)>);
+
+  static_assert(ranges::bidirectional_range<decltype(input)>);
+  static_assert(ranges::bidirectional_range<decltype(output)>);
+
+  static_assert(ranges::random_access_range<decltype(input)>);
+  static_assert(ranges::random_access_range<decltype(output)>);
+
+  static_assert(ranges::contiguous_range<decltype(input)>);
+  static_assert(!ranges::contiguous_range<decltype(output)>);
+
+  static_assert(ranges::range<decltype(output)>);
+  static_assert(ranges::semiregular<decltype(output)>);
+  static_assert(ranges::enable_view<decltype(output)>);
+}
+
+TEST(TakeEvenlyView, ConceptChecksFromBidirectionalRange) {
+  auto input = std::list{1, 2, 3, 4, 5, 6, 7, 8, 9, 10};
+  auto output = beluga::views::take_evenly(input, 2);
+
+  static_assert(ranges::forward_range<decltype(input)>);
+  static_assert(ranges::forward_range<decltype(output)>);
+
+  static_assert(ranges::bidirectional_range<decltype(input)>);
+  static_assert(ranges::bidirectional_range<decltype(output)>);
+
+  static_assert(!ranges::random_access_range<decltype(input)>);
+  static_assert(!ranges::random_access_range<decltype(output)>);
+}
+
 TEST(TakeEvenlyView, NoElementsTakeZero) {
   const auto input = std::vector<int>{};
-  const auto output = input | beluga::views::take_evenly(0) | ranges::to<std::vector>;
+  auto output = input | beluga::views::take_evenly(0);
```

Comment:
@nahueespinosa why not `const` anymore? Same elsewhere, except perhaps for random access tests.

Reply:
It comes from a time when the view cached something. I now used const when possible. There are only two exceptions:

1. `iota` can't be const as it is a range factory with mutable state.
2. `reverse` can't be const because it caches.